src/storage/btree.c

```diff
@@ -12222,7 +12224,11 @@ btree_find_oid_from_ovfl (THREAD_ENTRY * thread_p, BTID_INT * btid_int, PAGE_PTR
 	      *offset_to_object = CAST_BUFLEN (oid_ptr - ovf_record.data);
 	    }
 	  /* Not in this page. */
```

Comment:
@razvanradulescu 

I think I understand your intention. It seems you wanted `btree_seq_find_oid_from_ovfl` to check only the left side and the right side of oid and the above misses `return` for matched case.
```C
  if (is_match)
    {
      ...
      return NO_ERROR;
    }
```

I'd like to propose to remove matching from `btree_find_oid_from_ovfl` and change `btree_seq_find_oid_from_ovfl` to check matching from the `mid` position. So, the change would be as follows:
```C
  /* first, check previous OIDs */
  oid_ptr = initial_oid_ptr - obj_size;
```

->

```C
  /* first, check OID and previous ones */
  oid_ptr = initial_oid_ptr;
```